module/zfs/arc.c

```diff
@@ -5611,6 +5617,50 @@ arc_access(arc_buf_hdr_t *hdr, kmutex_t *hash_lock)
 	}
 }
 
+/*
+ * This routine is called by dbuf_hold() to update the arc_access() state
+ * which otherwise would be skipped for entries in the dbuf cache.
+ */
+void
+arc_buf_access(arc_buf_t *buf)
+{
+	mutex_enter(&buf->b_evict_lock);
+	arc_buf_hdr_t *hdr = buf->b_hdr;
+
+	/*
+	 * Avoid taking the hash_lock when possible as an optimization.
+	 * The header must be checked again under the hash_lock in order
+	 * to handle the case where it is concurrently being released.
+	 */
+	if (hdr->b_l1hdr.b_state == arc_anon || HDR_EMPTY(hdr)) {
+		mutex_exit(&buf->b_evict_lock);
```

Comment:
Any reason why `arcstat_access_skip` is not being bumped here? It's the same condition as below and results in an early exit.

Reply:
In my local testing I did add an additional kstat here in order to determine if including this additional check was worthwhile.  Keep in mind this check _could_ be dropped entirely since it's safe to rely on the one under the mutex.  Based on my  test results 99.99+% of the released headers were caught here without the need to take the hash lock, so I opted to keep it as an optimization to minimize any potential lock contention.  After determining that I was only really interested in how many slipped through since that was the exceptional case.  So the value is only bumped in the second conditional.